core/src/main/java/org/seedstack/seed/core/Seed.java

```diff
@@ -177,6 +203,46 @@ synchronized private void unregisterDiagnosticManager(String name) {
         diagnosticManagers.remove(name);
     }
 
+    private String getBanner() {
+        InputStream bannerStream = SeedReflectionUtils.findMostCompleteClassLoader(Seed.class).getResourceAsStream("banner.txt");
+        if (bannerStream != null) {
+            try {
+                return new Scanner(bannerStream).useDelimiter("\\Z").next();
+            } finally {
+                try {
+                    bannerStream.close();
+                } catch (IOException e) {
+                    // nothing to do
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private void checkConsistency(SeedRuntime seedRuntime) {
+        Set<String> inconsistentPlugins = new HashSet<String>();
+        String seedVersion = seedRuntime.getVersion();
+
+        if (seedVersion != null) {
+            for (Plugin plugin : ServiceLoader.load(Plugin.class)) {
```

Comment:
You can access the plugin classes via the Kernel, it might not be a good idea to load all the plugins twice. However good idea this check.